rllib/env/multi_agent_env.py

```diff
@@ -1,25 +1,39 @@
+import abc
 import gym
-from typing import Callable, Dict, List, Tuple, Type, Optional, Union
+import logging
+from typing import Callable, Dict, List, Tuple, Type, Optional, Union, Set
 
 from ray.rllib.env.base_env import BaseEnv
 from ray.rllib.env.env_context import EnvContext
-from ray.rllib.utils.annotations import ExperimentalAPI, override, PublicAPI
+from ray.rllib.utils.annotations import ExperimentalAPI, override, PublicAPI, \
+    DeveloperAPI
 from ray.rllib.utils.typing import AgentID, EnvID, EnvType, MultiAgentDict, \
     MultiEnvDict
 
 # If the obs space is Dict type, look for the global state under this key.
 ENV_STATE = "state"
 
+logger = logging.getLogger(__name__)
+
 
 @PublicAPI
-class MultiAgentEnv(gym.Env):
+class MultiAgentEnv(abc.ABC):
```

Comment:
what's abc.ABC?

Reply:
"Abstract base class".

This change causes some (minor) problems in the env checking function: `rollout_worker.py::_validate_env()`.
E.g. we need to add MultiAgentEnv to the list of allowed env types.

Reply:
I've modified this to now inherit from abc.ABC and gym.Env. 

I made it inherit from abc.ABC because I didn't want users to be able to instantiate this class as a standalone.